platforms/software/reporting/src/main/java/org/gradle/api/reporting/ReportingExtension.java

```diff
@@ -49,91 +52,48 @@ public abstract class ReportingExtension {
-        baseDirectory.set(project.getLayout().getBuildDirectory().dir(DEFAULT_REPORTS_DIR_NAME));
+        getBaseDirectory().convention(project.getLayout().getBuildDirectory().dir(DEFAULT_REPORTS_DIR_NAME));
     }
 
     /**
-     * The base directory for all reports
-     * <p>
-     * This value can be changed, so any files derived from this should be calculated on demand.
-     *
-     * @return The base directory for all reports
-     */
-    @ToBeReplacedByLazyProperty
-    public File getBaseDir() {
-        return baseDirectory.getAsFile().get();
-    }
-
-    /**
-     * Sets the base directory to use for all reports
-     *
-     * @param baseDir The base directory to use for all reports
-     * @since 4.0
-     */
-    public void setBaseDir(File baseDir) {
-        baseDirectory.set(baseDir);
-    }
-
-    /**
-     * Sets the base directory to use for all reports
-     * <p>
-     * The value will be converted to a {@code File} on demand via {@link Project#file(Object)}.
-     *
-     * @param baseDir The base directory to use for all reports
-     */
-    public void setBaseDir(final Object baseDir) {
-        this.baseDirectory.set(project.provider(new Callable<Directory>() {
-            @Override
-            public Directory call() throws Exception {
-                DirectoryProperty result = project.getObjects().directoryProperty();
-                result.set(project.file(baseDir));
-                return result.get();
-            }
-        }));
-    }
-
-    /**
-     * Returns base directory property to use for all reports.
+     * The base directory property to use for all reports.
      *
      * @since 4.4
      */
-    public DirectoryProperty getBaseDirectory() {
-        return baseDirectory;
-    }
+    @ReplacesEagerProperty(adapter = BaseDirAdapter.class)
+    public abstract DirectoryProperty getBaseDirectory();
 
     /**
-     * Creates a file object for the given path, relative to {@link #getBaseDir()}.
+     * Creates a file object for the given path, relative to {@link #getBaseDirectory()}.
      * <p>
      * The reporting base dir can be changed, so users of this method should use it on demand where appropriate.
      *
      * @param path the relative path
-     * @return a file object at the given path relative to {@link #getBaseDir()}
+     * @return a file object at the given path relative to {@link #getBaseDirectory()}
      */
     public File file(String path) {  // TODO should this take Object?
-        return this.project.getServices().get(FileLookup.class).getFileResolver(getBaseDir()).resolve(path);
+        return this.project.getServices().get(FileLookup.class).getFileResolver(getBaseDirectory().getAsFile().get()).resolve(path);
     }
 
-    @ToBeReplacedByLazyProperty
     // TODO this doesn't belong here, that java plugin should add an extension to this guy with this
-    public String getApiDocTitle() {
+    @ReplacesEagerProperty
```

Comment:
Ideally, this should be refactored, but it has been there for too long, and I'm not sure where it should go. I think it's outside the scope of this PR to fix that.

Reply:
Yeah, definitely this is out of scope of this PR, thanks for upgrading it anyway!